lib/compiler/src/compile.erl

```diff
@@ -673,6 +674,7 @@ standard_passes() ->
      {iff,debug_info,?pass(save_abstract_code)},
 
      ?pass(expand_records),
+     ?pass(collect_records),
```

Comment:
Just traversing the top-level terms of the AST to collect records should be fairly cheap. I doubt that complicating `erl_expand_records` would be worth it.